src/Microsoft.Extensions.Primitives/CompositeChangeToken.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace Microsoft.Extensions.Primitives
+{
+    /// <summary>
+    /// An <see cref="IChangeToken"/> which represents one or more <see cref="IChangeToken"/> instances.
+    /// </summary>
+    public class CompositeChangeToken : IChangeToken
+    {
+        private static readonly Action<object> _onChangeDelegate = OnChange;
+        public CancellationTokenSource _cancellationTokenSource;
+        private readonly object _callbackLock = new object();
+        private bool _registeredCallbackProxy;
+        private List<IDisposable> _disposables;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="CompositeChangeToken"/>.
+        /// </summary>
+        /// <param name="changeTokens">The list of <see cref="IChangeToken"/> to compose.</param>
+        public CompositeChangeToken(IReadOnlyList<IChangeToken> changeTokens)
+        {
+            if (changeTokens == null)
+            {
+                throw new ArgumentNullException(nameof(changeTokens));
+            }
+
+            ChangeTokens = changeTokens;
+
+            ActiveChangeCallbacks = false;
+            for (var i = 0; i < ChangeTokens.Count; i++)
+            {
+                if (ChangeTokens[i].ActiveChangeCallbacks)
+                {
+                    ActiveChangeCallbacks = true;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Returns the list of <see cref="IChangeToken"/> which compose the current <see cref="CompositeChangeToken"/>.
+        /// </summary>
+        public IReadOnlyList<IChangeToken> ChangeTokens { get; }
+
+        /// <inheritdoc />
+        public IDisposable RegisterChangeCallback(Action<object> callback, object state)
+        {
+            EnsureCallbacksInitialized();
+            if (!_cancellationTokenSource.IsCancellationRequested)
+            {
+                return _cancellationTokenSource.Token.Register(callback, state);
+            }
+
+            else
+            {
+                return null;
```

Comment:
The `if (!_cancellationTokenSource.IsCancellationRequested)` check ensures that if OnChange has been called, it will not Register the callback again. What else can I do here?